CesiumGltfContent/include/CesiumGltfContent/GltfUtilities.h

```diff
@@ -191,5 +195,44 @@ struct CESIUMGLTFCONTENT_API GltfUtilities {
    * @param bufferIndex The index of the buffer to compact.
    */
   static void compactBuffer(CesiumGltf::Model& gltf, int32_t bufferIndex);
+
+  /**
+   * @brief Hit result data for intersectRayGltfModelParametric
+   */
+  struct IntersectResult {
+    struct Hit {
+      glm::dvec3 primitivePoint = {};
+      glm::dmat4x4 primitiveToWorld = {};
+
+      glm::dvec3 worldPoint = {};
+      double rayToWorldPointDistanceSq = -1;
+
+      int meshId = -1;
+      int primitiveId = -1;
+    };
+
+    std::optional<Hit> hit;
+
+    std::vector<std::string> warnings{};
+  };
+
+  /**
+   * @brief Intersects a ray with a glTF model and returns the first
+   * intersection point.
+   *
+   * Supports all mesh primitive modes.
+   * Points and lines are assumed to have no area, and are ignored
+   *
+   * @param ray A ray in world space.
+   * @param gltf The glTF model to intersect.
+   * @param cullBackFaces Ignore triangles that face away from ray.
+   * @param gltfTransform Optional matrix to apply to entire gltf model.
+   * @param return IntersectResult describing outcome
+   */
+  static IntersectResult intersectRayGltfModel(
```

Comment:
Should we provide a function that returns all of the intersections, not just the first one? A common use-case for that is ignoring hits against translucent geometry.

Reply:
That's a great idea, but I think I'd like to write a new issue for it.

This PR is already fairly big. But also, I wouldn't want to short change the "translucent geometry" use case. It could use some unit tests and testing against Ion assets. It may even overlap with that earlier comment about supporting double sided materials.

Reply:
Ok, that's fine. I'm slightly concerned about the backward compatibility implications of adding it later, but I guess it could be a separate function anyway, like `intersectRayGltfModelMultiple`?